src/hdmf_zarr/zarr_utils.py

```diff
@@ -73,6 +74,14 @@ def __iter__(self):
     def __next__(self):
         return self._get_ref(super().__next__())
 
+    def append(self, arg):
+        # Create Builder
+        builder = self.io.manager.build(arg)
+
+        # Create ZarrReference
+        ref = self.io._create_ref(builder)
```

Comment:
Note: you had `self.io._ZarrIO.__get_ref(builder)` which goes against the convention that private methods are not called outside of its class. This makes refactoring harder. It's totally valid to call this method here, but then I think we should rename it to have a single underscore (and also rename to create instead of get, which implies it is just retrieving an object).